students/Thai_H/lesson4/mail_room_part2.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+import sys
+import os
+
+#--------------------------------------------------------------------------
+def send_thank_you_letter(donor_name, donation_amt):
+    print(f'\nDear {donor_name}\n'
+         'Thank you for a recent donation to our Foundation in the amount of $' + f'{donation_amt}\n'
+         '\nSincerely'
+         '\nBill & Melinda Gates')
+
+
+#---------------------------------------------------------------------------
+def is_existing_donor(donor_name):
+    for each_donor in list_of_donors:
+        if each_donor['name'] == donor_name:
+            return True  # returns True if existing donor
+    #--- after finished looping all donors and hence have not returned, then NEW donor
+    return False #False if new donor
+
+
+#-----------------------------------------------------------------------------
+def send_thank_you_add_new_donation():
+    donor_name =input('\n\nPlease enter name of donor:  ')
+    donation_amt = input('\nEnter the amount > ')
+    if is_existing_donor(donor_name):
+        for each_donor in list_of_donors:
+            if donor_name == each_donor['name']:  #[0] is existing donor name
+                each_donor['donation'].append(float(donation_amt))
+
+    # New donor
+    else:
+        new_donor_name_amt = {'name':donor_name,'donation':[float(donation_amt)]}
+        list_of_donors.append(new_donor_name_amt)
+    # send thank you letter to the the donor name user just entered
+    send_thank_you_letter(donor_name, donation_amt)
```

Comment:
This doesn't appear to be functioning correctly - isn't adding new donors or new donations to existing amounts.